kplutus-pyk/src/kplutus_pyk/__main__.py

```diff
@@ -1,36 +1,67 @@
-from argparse import ArgumentParser
+import sys
+from argparse import ArgumentParser, ArgumentTypeError
+from pathlib import Path
 
 from pyk.cli_utils import dir_path, file_path
 
 from .kplutus import KPlutus
 
 
 def main() -> None:
+    sys.setrecursionlimit(15000000)
+
     parser = create_parser()
     args, remainder = parser.parse_known_args()
 
     if args.command == 'kompile':
         KPlutus.kompile(args.main_file, remainder)
     if args.command == 'uplc-to-k':
-        KPlutus.uplc_to_k(args.main_file, args.definition, remainder)
+        kplutus = KPlutus(args.kplc_lib)
+        if not args.definition:
+            args.definition = kompiled_dir(".")
+        kplutus.uplc_to_k(args.main_file, args.definition, remainder)
     else:
         raise AssertionError()
 
 
 def create_parser() -> ArgumentParser:
-    parser = ArgumentParser()
+    kplc_args = ArgumentParser(add_help=False)
+    kplc_args.add_argument('--kplc-lib', type=dir_path, help='Path to the kplutus library')
+
+    k_args = ArgumentParser(add_help=False)
+    ex = k_args.add_mutually_exclusive_group()
+    ex.add_argument('--definition', type=dir_path, dest='definition', help='Path to exact kompiled directory.')
+    ex.add_argument(
+        '--directory', type=kompiled_dir, dest='definition', help='Path to where the kompiled directory is.'
```

Comment:
Consider removing this argument. Flag `--definition` is more direct and more general, so it should be the preferred way of passing the kompiled definition (`--directory` is there for K tools for backwards compatibility, but this does not seem to apply here).

Reply:
For now, `kplc` only takes the `--directory` option, so to avoid confusion around how it passes it through to `kplutus-pyk` I've added it here.